PrimeNumbers/PrimeNumber_join.cpp

```diff
@@ -173,17 +240,23 @@ DWORD WINAPI ThreadWorker(LPVOID lpParam)
                 tInput->hardWaitWakeupTimeTicks += hardWaitWakeupLatency;
                 tInput->spinLoopTimeTicksHardWait += spinWaitCpuCycles;
                 tInput->hardWaitCount++;
+                tInput->totalIterationsHardWaits += iterations;
 
-                PRINT_HARD_WAIT_LATENCY("%d. %lld cycles, %llu total spin-loop cycles", threadId, i, hardWaitWakeupLatency, spinWaitCpuCycles);
+                //PRINT_HARD_WAIT_LATENCY("%d. wakeup latency %lld cycles, %llu total spin-loop cycles", threadId, i, hardWaitWakeupLatency, spinWaitCpuCycles);
+                //printf("thread %d - hardwait %d, total wait %d [%d out of %d]\n", threadId, tInput->hardWaitCount, (tInput->hardWaitCount + tInput->softWaitCount), tInput->processed, tInput->count);
             }
             else
             {
                 unsigned __int64 softWaitWakeupLatency = joinData->getTicksSinceRestart();
                 tInput->softWaitWakeupTimeTicks += softWaitWakeupLatency;
                 tInput->spinLoopTimeTicksSoftWait += spinWaitCpuCycles;
                 tInput->softWaitCount++;
+                tInput->totalIterationsSoftWaits += iterations;
 
-                PRINT_SOFT_WAIT_LATENCY("%d. %lld wake-up cycles, %llu total spin-loop cycles.", threadId, i, softWaitWakeupLatency, spinWaitCpuCycles);
+                //PRINT_SOFT_WAIT_LATENCY("%d. wakeup latency %lld cycles, %llu total spin-loop cycles.", threadId, i, softWaitWakeupLatency, spinWaitCpuCycles);
```

Comment:
should we uncomment this?

Reply:
feel free to comment/uncomment any print stuff.